test/functional/terminal/tui_spec.lua

```diff
@@ -150,8 +151,11 @@ describe('tui', function()
   end)
 
   it('can handle focus events', function()
+    execute('set noshowmode')
```

Comment:
If this line isn't here then in insert mode the cmdline becomes `--INSERT--` instead of displaying any messages (such as `gained` or `lost`).

Echoing messages seemed like the simplest way of testing the events firing.